lib/chef-acceptance/application.rb

```diff
@@ -0,0 +1,57 @@
+require "chef-acceptance/output_formatter"
+require "chef-acceptance/chef_client_run"
+require "chef-acceptance/test_suite"
+require "chef-acceptance/executable_helper"
+
+module ChefAcceptance
+  # Responsible for running one or more suites and displaying statistics about
+  # the runs to the user
+  class Application
+    attr_reader :force_destroy
+
+    def initialize(options = {})
+      @force_destroy = options.fetch(:force_destroy, false)
+      #output_formatter = OutputFormatter.new
+      #suite_stats = SuiteStatistics.new
+    end
+
+    def run(suite, command)
+      test_suite = TestSuite.new(suite)
+
+      unless test_suite.exist?
+        fail <<-EOS.gsub(/^\s+/, "")
+          Could not find test suite '#{test_suite.name}' in the current working directory '#{Dir.pwd}'.
+        EOS
+      end
+
+      unless ExecutableHelper.executable_installed? 'chef-client'
+        fail "Could not find chef-client in #{ENV['PATH']}"
+      end
+
+      if command == "test"
+        AcceptanceCookbook::CORE_ACCEPTANCE_RECIPES.each do |recipe|
+          begin
+            run_command(test_suite, recipe)
+          rescue RuntimeError
+            if force_destroy && recipe != "destroy"
```

Comment:
maybe should should raise a message like "destroy recipe failed.  Ignoring force destroy"